### ISSUE TITLE
FP derivable_impls:  attributes are ignored

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#11368]
### Summary

When clippy suggests deriving an impl, it still warn when the fn fulfilling the impl has some special attribute, such as `inline`, `cold` or `track_caller`.

When we derive the impl, we no longer get the desired attribute effect.

### Lint Name

derivable_impls

### Reproducer

I tried this code:

```rust
struct A {
    a: u32
}

impl  Default for A {
    #[track_caller]
    fn default() -> A {
        A {
            a: 0,
        }
    }
}
```

I expected to see this happen:
No warning

Instead, this happened:
````
warning: this `impl` can be derived
  --> src/main.rs:10:1
   |
10 | / impl  Default for A {
11 | |     #[track_caller]
12 | |     fn default() -> A {
13 | |         A {
...  |
16 | |     }
17 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derivable_impls
   = note: `#[warn(clippy::derivable_impls)]` on by default
   = help: remove the manual implementation...
help: ...and instead derive it
   |
6  + #[derive(Default)]
7  | struct A {
   |
````

### Version

```text
rustc 1.73.0-nightly (6ef7d16be 2023-08-19)
binary: rustc
commit-hash: 6ef7d16be0fb9d6ecf300c27990f4bff49d22d46
commit-date: 2023-08-19
host: x86_64-unknown-linux-gnu
release: 1.73.0-nightly
LLVM version: 17.0.0
```

**Comment by y21** [comment]
This has the wrong label, right? Should be FP not FN.
Let's see if I have permissions to change labels
@rustbot label I-false-positive -I-false-negative

**Comment by y21** [comment]
This was fixed in #11404 so I think it can be closed
(cc @mojave2 you can write `Fixes #xxxxx` in a PR description to autoclose an issue when the PR gets merged)
